spec/controllers/miq_report_controller/dashboards_spec.rb

```diff
@@ -1,27 +1,57 @@
 describe ReportController do
   context "::Dashboards" do
-    context "#db_edit" do
-      let(:user) { FactoryGirl.create(:user, :features => "db_edit") }
-      before :each do
-        login_as user
-        @db = FactoryGirl.create(:miq_widget_set,
-                                 :owner    => user.current_group,
-                                 :set_data => {:col1 => [], :col2 => [], :col3 => []})
-      end
+    let(:miq_widget_set) { FactoryGirl.create(:miq_widget_set, :owner => user.current_group, :set_data => {:col1 => [], :col2 => [], :col3 => []}) }
+    let(:user)           { FactoryGirl.create(:user, :features => "db_edit") }
+
+    before do
+      login_as user
+    end
 
+    describe "#db_edit" do
       it "dashboard owner remains unchanged" do
         allow(controller).to receive(:db_fields_validation)
         allow(controller).to receive(:replace_right_cell)
-        owner = @db.owner
+        owner = miq_widget_set.owner
         new_hash = {:name => "New Name", :description => "New Description", :col1 => [1], :col2 => [], :col3 => []}
         current = {:name => "New Name", :description => "New Description", :col1 => [], :col2 => [], :col3 => []}
-        controller.instance_variable_set(:@edit, :new => new_hash, :db_id => @db.id, :current => current)
-        controller.instance_variable_set(:@_params, :id => @db.id, :button => "save")
+        controller.instance_variable_set(:@edit, :new => new_hash, :db_id => miq_widget_set.id, :current => current)
+        controller.instance_variable_set(:@_params, :id => miq_widget_set.id, :button => "save")
         controller.db_edit
-        expect(@db.owner.id).to eq(owner.id)
+        expect(miq_widget_set.owner.id).to eq(owner.id)
         expect(assigns(:flash_array).first[:message]).to include("saved")
         expect(controller.send(:flash_errors?)).not_to be_truthy
       end
     end
+
+    describe "#db_save_members" do
+      let(:miq_widget_col_1_1) { FactoryGirl.create(:miq_widget) }
+      let(:miq_widget_col_1_2) { FactoryGirl.create(:miq_widget) }
+      let(:miq_widget_col_2_1) { FactoryGirl.create(:miq_widget) }
+      let(:miq_widget_col_2_2) { FactoryGirl.create(:miq_widget) }
+      let(:miq_widget_col_3_1) { FactoryGirl.create(:miq_widget) }
+      let(:miq_widget_col_3_2) { FactoryGirl.create(:miq_widget) }
+      let(:set_data) do
+        {:col1 => [miq_widget_col_1_1.id, miq_widget_col_1_2.id],
+         :col2 => [miq_widget_col_2_1.id, miq_widget_col_2_2.id],
+         :col3 => [miq_widget_col_3_1.id, miq_widget_col_3_2.id]}
+      end
```

Comment:
This could be rewritten to something like this:
```ruby
let(:set_data) do
  3.times.map { |n| ["col#{n+1}".to_sym, Array.new(2, FactoryGirl.create(:miq_widget))] }.to_h
end
```
Or maybe differently, but please don't use `let` blocks as you use them above.